src/libraries/System.Text.Json/docs/collection_support.md

```diff
@@ -0,0 +1,158 @@
+   [`IList<T>`](https://docs.microsoft.com/dotnet/api/system.collections.generic.ilist-1?view=netcore-3.1),
+   [`ICollection<T>`](https://docs.microsoft.com/dotnet/api/system.collections.generic.icollection-1?view=netcore-3.1),
+   [`IDictionary`](https://docs.microsoft.com/dotnet/api/system.collections.idictionary?view=netcore-3.1),
+   [`IDictionary<string, TValue>`](https://docs.microsoft.com/dotnet/api/system.collections.generic.idictionary-2?view=netcore-3.1),
+   [`Stack<T>`](https://docs.microsoft.com/dotnet/api/system.collections.generic.stack-1?view=netcore-3.1)\*,
+   [`Queue<T>`](https://docs.microsoft.com/dotnet/api/system.collections.generic.queue-1?view=netcore-3.1),
+   [`ConcurrentStack<T>`](https://docs.microsoft.com/dotnet/api/system.collections.concurrent.concurrentstack-1?view=netcore-3.1)\*,
+   [`ConcurrentQueue<T>`](https://docs.microsoft.com/dotnet/api/system.collections.concurrent.concurrentqueue-1?view=netcore-3.1),
+   [`Stack`](https://docs.microsoft.com/dotnet/api/system.collections.stack?view=netcore-3.1)\*,
+   and [`Queue`](https://docs.microsoft.com/dotnet/api/system.collections.queue?view=netcore-3.1)
+4. The element type is supported by [`JsonSerializer`](https://docs.microsoft.com/dotnet/api/system.text.json.jsonserializer?view=netcore-3.1)
+
+All custom collections (everything that derives from [`IEnumerable`](https://docs.microsoft.com/dotnet/api/system.collections.ienumerable?view=netcore-3.1))
+are supported for serialization, as long as their element types are supported.
+
+There are known issues with some custom collections where we don't offer round-trippable support.
+These include:
+
+- Support for [`ExpandoObject`](https://docs.microsoft.com/dotnet/api/system.dynamic.expandoobject?view=netcore-3.1): https://github.com/dotnet/corefx/issues/38007
+- Support for [`DynamicObject`](https://docs.microsoft.com/dotnet/api/system.dynamic.dynamicobject?view=netcore-3.1): https://github.com/dotnet/corefx/issues/41105
+- Support for [`DataTable`](https://docs.microsoft.com/dotnet/api/system.data.datatable?view=netcore-3.1): https://github.com/dotnet/corefx/issues/38712
+- Support for [`FormFile`](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.http.formfile?view=aspnetcore-3.1): https://github.com/dotnet/corefx/issues/41401
+- Support for [`IFormCollection`](https://docs.microsoft.com/dotnet/api/microsoft.aspnetcore.http.iformcollection?view=aspnetcore-3.1)
+- Assigning `null` to value-type collections like [`ImmutableArray<T>`](https://docs.microsoft.com/dotnet/api/system.collections.immutable.immutablearray-1?view=netcore-3.1): https://github.com/dotnet/corefx/issues/42399
+
+For more information, see the [open issues in System.Text.Json](https://github.com/dotnet/runtime/issues?q=is%3Aopen+is%3Aissue+label%3Aarea-System.Text.Json).
+
+\* All instances of custom collections that implement stack types are reversed on serialization,
```

Comment:
Was repeating this intentional?

Reply:
Yes, there was a distinction between built-in types and custom types. I've consolidated these notes at the bottom.